uuid/v3.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+import { bytesToUuid, uuidToBytes } from "./_common.ts";
+import { concat } from "../bytes/concat.ts";
+import { assert } from "../_util/asserts.ts";
+import { crypto } from "../crypto/crypto.ts";
+
+const UUID_RE =
+  /^[0-9a-f]{8}-[0-9a-f]{4}-[3][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+
+/**
+ * Validate that the passed UUID is an RFC4122 v3 UUID.
+ *
+ * @example
+ * ```ts
+ * import { generate as generateV3, validate } from "https://deno.land/std@$STD_VERSION/uuid/v3.ts";
+ *
+ * validate(await generateV3("6ba7b810-9dad-11d1-80b4-00c04fd430c8", new Uint8Array())); // true
+ * validate(crypto.randomUUID()); // false
+ * validate("this-is-not-a-uuid"); // false
+ * ```
+ */
+export function validate(id: string): boolean {
+  return UUID_RE.test(id);
+}
+
+/**
+ * Generate a RFC4122 v3 UUID (MD5 namespace).
+ *
+ * @example
+ * ```js
+ * import { generate } from "https://deno.land/std@$STD_VERSION/uuid/v3.ts";
+ *
+ * const NAMESPACE_URL = "6ba7b810-9dad-11d1-80b4-00c04fd430c8";
```

Comment:
This UUID looks like NAMESPACE_DNS according to RFC4122 appendix C ref. https://datatracker.ietf.org/doc/html/rfc4122#appendix-C (NAMESPACE_URL is `6ba7b811-9dad-11d1-80b4-00c04fd430c8` instead. You can also check it with python's `uuid` standard module.

```shellsession
$ python3
Python 3.11.2 (main, Feb 16 2023, 02:51:42) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import uuid
>>> uuid.NAMESPACE_DNS
UUID('6ba7b810-9dad-11d1-80b4-00c04fd430c8')
>>> uuid.NAMESPACE_URL
UUID('6ba7b811-9dad-11d1-80b4-00c04fd430c8')
```

Reply:
Just wanted to note that I got that example from the UUIDv5 implementation... this PR fixes this mistake over there as well.